crypto/signature.go

```diff
@@ -17,33 +17,48 @@ type StablyMarshallable interface {
 //
 // Signature is mutually compatible with github.com/nspcc-dev/neofs-api-go/v2/refs.Signature
 // message. See ReadFromV2 / WriteToV2 methods.
-//
-// Note that direct typecast is not safe and may result in loss of compatibility:
-//
-//	_ = Signature(refs.Signature{}) // not recommended
-type Signature refs.Signature
+type Signature struct {
```

Comment:
I'm still not sure what exactly this solves. Code samples?

Reply:
allows to provide `func (Signature) PublicKey() PublicKey` without error. Also disables ability to do `Signature(refs.Signature{})` casts

Reply:
Not worth the trouble, I'd say. All of these "must not be called" will be happily ignored in most of the cases.

Reply:
as a dev, i would be glad if i have a certain info source that says me how should a struct be inited correctly: e.g. i am going to use `Signature` and the comments to it says `Use [Calculate] or [ReadFromV2] `

Reply:
Having a constructor for the struct with logic - is an understandable way. I vote for constructors for all _logic_ structs.
Data structs maybe initiated as is inline, I mean
```go
a := SomeStruct{
    A:10,
    B:"name",
...
}
```

Reply:
constructors are ok (not this PR theme), struct initialization may be hard to support in terms of compatibility (the main power of methods/constructors)

Reply:
> as a dev, i would be glad if i have a certain info source that says me how should a struct be inited correctly: e.g. i am going to use `Signature` and the comments to it says `Use [Calculate] or [ReadFromV2] `

if you read docs, u'll get it anyway. I guess ur more for constructors

Reply:
created #483 for the future

Reply:
I still don't understand this change, we're creating another wrapper for... what?

I see there is some code in neofs-node that wants to extract public keys from signature. OK, can be provided with a new method.

Then there is some creating a signature from scheme/key/value. OK, can be provided with `New(scheme, key, value)`.

But why a wrapper that raises inevitable questions of init and sync to the lower level structure?

Reply:
i don't create any wrapper, there is a type `Signature` and its interface. If we agree with interface, then that's how we implement it. Besides this, we're willing to get rid of api-go intermediate types https://github.com/nspcc-dev/neofs-api-go/issues/395, so, `type Signature refs.Signature` obviously not how  we wanna define it

Reply:
so, we may discard `PublicKey() neofscrypto.PublicKey` method and leave type definition untouched. Right now it's not used in node, do we think it may be useful? Binary getter is needed at least for printing